src/cargo/util/lints.rs

```diff
@@ -880,6 +882,111 @@ pub fn unused_dependencies(
+            "".to_string()
+        };
+
+        let mut emit_lint = |msg| match lint_level {
+            LintLevel::Warn => gctx.shell().warn(msg),
+            LintLevel::Deny | LintLevel::Forbid => {
+                *error_count += 1;
+                gctx.shell().error(msg)
+            }
+            LintLevel::Allow => Ok(()),
+        };
+
+        cfg_expr.try_for_each(|cfg| -> CargoResult<()> {
+            let (name, value) = match cfg {
+                Cfg::Name(name) => (name, None),
+                Cfg::KeyPair(name, value) => (name, Some(value.to_string())),
+            };
+
+            match check_cfg.expecteds.get(name) {
+                Some(ExpectedValues::Some(values)) if !values.contains(&value) => {
+                    let value = if let Some(ref value) = value {
+                        value
+                    } else {
+                        "(none)"
+                    };
+                    emit_lint(format!(
+                        "{prefix}unexpected `cfg` condition value: `{value}` for `{cfg}` in `[target.'cfg({cfg_expr})'{suffix}]`"
+                    ))?;
+                }
+                None => {
+                    emit_lint(format!(
+                        "{prefix}unexpected `cfg` condition name: `{name}`{cfg} in `[target.'cfg({cfg_expr})'{suffix}]`",
+                        cfg = if value.is_some() {
+                            format!(" for `{cfg}`")
+                        } else {
+                            format!("")
+                        }
+                    ))?;
+                }
+                _ => { /* not unexpected */ }
+            }
+            Ok(())
+        })
+    }
+
+    let rustc = gctx.load_global_rustc(Some(ws))?;
+    // FIXME: While it doesn't doesn't really matter for `--print=check-cfg`, wee should
+    // still pass the actual requested targets instead of an empty array so that the
+    // target info can be de-duplicated.
+    let compile_kinds = CompileKind::from_requested_targets(gctx, &[])?;
+    let target_info = TargetInfo::new(gctx, &compile_kinds, &rustc, CompileKind::Host)?;
+
+    let Some(global_check_cfg) = target_info.check_cfg() else {
+        return Ok(());
+    };
+
+    for dep in pkg.summary().dependencies() {
+        let Some(platform) = dep.platform() else {
+            continue;
+        };
+        let Platform::Cfg(cfg_expr) = platform else {
+            continue;
+        };
+
+        // FIXME: If the `[lints.rust.unexpected_cfgs.check-cfg]` config is set we should
+        // re-fetch the check-cfg informations with those extra args
+
+        if !global_check_cfg.exhaustive {
+            continue;
+        }
+
+        warn_on_unexpected_cfgs(
+            gctx,
+            &global_check_cfg,
+            cfg_expr,
+            // FIXME: We should get the lint level from `[lints.rust.unexpected_cfgs]`
+            LintLevel::Warn,
+            error_count,
+            Some(path),
+            ".dependencies",
```

Comment:
1. Why is this passed in rather than hard coded?
2. Why is it passed in in a dotted form?

Reply:
These were coming from a previous iteration of this PR. Fixed.